src/extension.ts

```diff
@@ -4,11 +4,35 @@ import * as child_process from 'child_process';
 import * as vscode from 'vscode';
 import * as languageclient from 'vscode-languageclient';
 import * as path from 'path';
+import * as fs from 'fs';
+import { WorkspaceFolder, DebugConfiguration, ProviderResult, CancellationToken } from 'vscode';
 
+function findQoreScript(context: vscode.ExtensionContext, scriptName: string): string {
+    if (path.isAbsolute(scriptName)) {
+        return scriptName;
+    }
+    // try extension directory
+    let s = path.join(context.extensionPath, scriptName);
+    if (fs.existsSync(s)) {
+        return s;
+    }
+    // try PATH environment variable
+    let pathArr = (process.env.PATH || "").split(path.delimiter);
+    for (const p of pathArr) {
+        s = path.join(p, scriptName);
+        if (fs.existsSync(s)) {
+            return s;
+        }
+    }
+    return scriptName;
+}
+
+// tutorial abouut "=>" https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions
 export async function activate(context: vscode.ExtensionContext) {
     console.log('Activating qore-vscode extension');
 
     let qore_executable: string = vscode.workspace.getConfiguration("qore").get("executable") || "qore";
+    console.log('Qore executable: '+qore_executable);
 
     // Find out if Qore and the astparser module are present.
```

Comment:
maybe we should add a check here, whether the qore bin is the requested version?

Reply:
It's a little thing, but `let` should only be used if you plan on re-declaring the variable. If the variable does not change it should be `const`.